### PR TITLE
Task 2

### PR DESCRIPTION
**Pull request by anastasiyakoush**

### CODE REVIEW

File: react/counter-app/src/containers/CounterContainer.js [hunk 1]
```diff
@@ -5,29 +5,71 @@ class CounterContainer extends Component {
...
-      count: 0
+      count: props.count,
+      id: props.id
     };
   }
 
   incrementCount = () => {
-    this.setState({ count: this.state.count + 1 });
+    this.setState(state => {
+      state = { ...state, count: state.count + 1 };
```
Comment (hellsatsu) [inline react/counter-app/src/containers/CounterContainer.js]: https://ru.reactjs.org/docs/state-and-lifecycle.html
Avoid to assign implicitly state.

File: react/counter-app/src/containers/CounterContainer.js [hunk 2]
```diff
@@ -5,29 +5,71 @@ class CounterContainer extends Component {
...
+      this.props.updateParent(state.id, state.count);
+      return state;
+    });
   };
 
   reset = () => {
-    this.setState({ count: 0 });
+    this.setState(state => {
+      state = { ...state, count: 0 };
```
Comment (hellsatsu) [inline react/counter-app/src/containers/CounterContainer.js]: same here

File: react/counter-app/src/containers/ParentContainer.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
+import React, { Component } from "react";
+import { Container, Box } from "@material-ui/core";
+import { ThemeProvider } from "@material-ui/styles";
+import Grid from "@material-ui/core/Grid";
+import Divider from "@material-ui/core/Divider";
+import CounterContainer from "./CounterContainer";
+import Counter from "../components/Counter/index";
+import theme from "../components/Counter/styles";
+
+const MIN_COUNT = 1;
```
Comment (mxbelski) [inline react/counter-app/src/containers/ParentContainer.js]: put all global consts into "const.js" file

File: react/counter-app/src/containers/ParentContainer.js [hunk 4]
```diff
@@ -0,0 +1,118 @@
+import React, { Component } from "react";
+import { Container, Box } from "@material-ui/core";
+import { ThemeProvider } from "@material-ui/styles";
+import Grid from "@material-ui/core/Grid";
+import Divider from "@material-ui/core/Divider";
+import CounterContainer from "./CounterContainer";
+import Counter from "../components/Counter/index";
```
Comment (mxbelski) [inline react/counter-app/src/containers/ParentContainer.js]: import position is wrong.

File: react/counter-app/src/containers/ParentContainer.js [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+  componentWillReceiveProps(nextProps) {
+    console.log(
+      `ParentContainer ${this.state.id} -- componentWillReceiveProps`
+    );
+    if (this.state.count !== nextProps.count) {
+      this.setState({ ...nextProps });
+    }
+  }
+  shouldComponentUpdate() {
```
Comment (mxbelski) [inline react/counter-app/src/containers/ParentContainer.js]: empty space

File: react/counter-app/src/containers/ParentContainer.js [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+import { ThemeProvider } from "@material-ui/styles";
+import Grid from "@material-ui/core/Grid";
+import Divider from "@material-ui/core/Divider";
+import theme from "../components/Counter/styles";
+
+class ParentContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      count: MIN_COUNT,
```
Comment (hellsatsu) [inline react/counter-app/src/containers/ParentContainer.js]: please, use more meaningful name to constant (example: MIN_COUNTER_AMOUNT)

**Pull request merged**